### ISSUE TITLE
Improvements to `bbhnet.injection.generate_gw`

### ISSUE THREAD

**Issue by alecgunny** [ML4GW/BBHNet#70]
- Accept `sample_rate` and `duration` arguments which are `Optional`. If `waveform_sampler` is not `None`, then these should be `None` (enforce this), otherwise use these arguments to instantiate a waveform sampler.
- Should `minimum_frequency` be an argument as well, with 20 as a default?
- How to dynamically infer the number of polarizations (to initialize the `signals` array) and ensure they are always concatenated in the same order?
- Can we do the filtering at the end once all the polarizations have been inserted into the `signals` array?

**Comment by EthanMarx** [comment]
Also, `waveform_approximant` as an option

**Comment by alecgunny** [comment]
@EthanMarx is that the `"IMRPhenomPv2"`?

The `sample_rate` and `waveform_duration` args should also be made explicit in `project_raw_gw` in lieu of the `waveform_generator` arg, which as far as I can tell is only used to capture these attributes. This will allow for a fix to the [TODO](https://github.com/ML4GW/BBHNet/blob/1ff6762853e5a15c8a2e13f193f57f0a764bca40/libs/data/bbhnet/data/waveform_sampler.py#L104) in `bbhnet.data.waveform_sampler.RandomWaveformDataset` to pass these arguments rather than instantiating a `_DummyWaveformGenerator` object

**Comment by EthanMarx** [comment]
@alecgunny Yes. We may in the future want to use waveforms with more complex physics and what not.

**Comment by deepchatterjeeligo** [comment]
Since this is marked as "good first issue", I can have a stab at it, will be a good opportunity to look at the source code. Unless this was urgent.

**Comment by EthanMarx** [comment]
Not urgent, definitely go for it.
